pedalboard/plugin_templates/PrimeWithSilence.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * pedalboard
+ * Copyright 2022 Spotify AB
+ *
+ * Licensed under the GNU Public License, Version 3.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "../JuceHeader.h"
+#include "../Plugin.h"
+#include "../plugins/AddLatency.h"
+#include <mutex>
+
+namespace Pedalboard {
+
+/**
+ * A dummy plugin that buffers audio data internally, used to test Pedalboard's
+ * automatic latency compensation.
+ */
+template <typename T, typename SampleType = float>
+class PrimeWithSilence
+    : public JucePlugin<juce::dsp::DelayLine<
+          SampleType, juce::dsp::DelayLineInterpolationTypes::None>> {
+public:
+  virtual ~PrimeWithSilence(){};
+
+  virtual void prepare(const juce::dsp::ProcessSpec &spec) {
+    JucePlugin<juce::dsp::DelayLine<
+        SampleType,
+        juce::dsp::DelayLineInterpolationTypes::None>>::prepare(spec);
+    plugin.prepare(spec);
+  }
+
+  virtual void reset() override {
+    this->getDSP().reset();
+    this->getDSP().setMaximumDelayInSamples(silenceLengthSamples);
+    this->getDSP().setDelay(silenceLengthSamples);
+    plugin.reset();
+    samplesOutput = 0;
+  }
+
+  virtual int
+  process(const juce::dsp::ProcessContextReplacing<float> &context) override {
+    this->getDSP().process(context);
+
+    // Context now has a delayed signal in it:
+    int samplesProcessed = plugin.process(context);
+    samplesOutput += samplesProcessed;
+
+    return std::max(
+        0, std::min((int)samplesProcessed,
+                    (int)samplesOutput - (int)this->getDSP().getDelay()));
+  }
+
+  virtual int getLatencyHint() override {
+    return this->getDSP().getDelay() + getNestedPlugin().getLatencyHint();
+  }
+
+  T &getNestedPlugin() { return plugin; }
+
+  void setSilenceLengthSamples(int newSilenceLengthSamples) {
+    this->getDSP().setMaximumDelayInSamples(newSilenceLengthSamples);
+    this->getDSP().setDelay(newSilenceLengthSamples);
+    silenceLengthSamples = newSilenceLengthSamples;
+
+    reset();
```

Comment:
Why is the reset necessary here? Wonder if should remove this set function in general and just force the creation of a new object?